src/Common/tests/System/Net/Http/Http2LoopbackServer.cs

```diff
@@ -515,6 +520,15 @@ public async Task<(int streamId, HttpRequestData requestData)> ReadAndParseReque
             requestData.Method = requestData.GetSingleHeaderValue(":method");
             requestData.Path = requestData.GetSingleHeaderValue(":path");
 
+            while ((frame.Flags & FrameFlags.EndStream) == 0)
+            {
+                frame = await ReadFrameAsync(Timeout).ConfigureAwait(false);
+                if (frame.Type == FrameType.Data && frame.Length > 1)
```

Comment:
This is going to discard any non-DATA frame. That seems bad.

Reply:
should we throw for now?